File: scripts/algorithms/Move Zeroes/Move Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int nums_size = nums.size();
        int cntr=0;
        
        for(int i=0;i<nums_size;i++)
        {
            if(nums[i] != 0)
            {
                nums[cntr] = nums[i];
                cntr++;
            }
        }
        for( int i=cntr;i<nums_size;i++)
        {
            nums[i] = 0;
        }
    }
};

File: algorithms/M/Move Zeroes/Move Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
       int a = 0, z = 0, temp;
        while(a < nums.length){
            if(nums[a] != 0){
                temp = nums[z];
                nums[z] = nums[a];
                nums[a] = temp; 
                z += 1;
            }
            a += 1;
        }
    }
}
